SELearning.Infrastructure/CommentManager.cs

```diff
@@ -1,38 +1,87 @@
-using SELearning.Core.Services;
-using SELearning.Core;
-using System;
+using SELearning.Core.Comment;
 
 namespace SELearning.Infrastructure
 {
     public class CommentManager : ICommentService
     {
-        public void PostComment(string author, string content)
+        ICommentRepository _repo;
+        public CommentManager(ICommentRepository repo)
         {
+            _repo = repo;
+        }
 
+        public async Task PostComment(CommentCreateDTO dto)
+        {
+            if ((await _repo.AddComment(dto)).Item1 == OperationResult.NotFound)
+            {
+                throw new Exception("The content that the comment belongs to could not be found");
+            }
         }
-        public void UpdateComment(Comment cmt)
+
+        public async Task UpdateComment(int id, CommentUpdateDTO dto)
         {
+            if ((await _repo.UpdateComment(id, dto)).Item1 == OperationResult.NotFound)
+            {
+                throw new Exception("The comment could not be found");
```

Comment:
Can we create a custom exception or something instead of constantly repeating this? Also to make catching this specific exception easier?

Reply:
This is already on my todo for tomorrow. I have talked to @YoHype as well, such that we will use the same exceptions for content and comment.